source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/TableCellRenderer.kt

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package io.adaptivecards.renderer.readonly
+
+import android.content.Context
+import android.graphics.Color
+import android.graphics.drawable.GradientDrawable
+import android.view.View
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import android.widget.TableLayout
+import android.widget.TableRow
+import androidx.core.view.updateLayoutParams
+import androidx.fragment.app.FragmentManager
+import io.adaptivecards.objectmodel.*
+import io.adaptivecards.renderer.*
+import io.adaptivecards.renderer.actionhandler.ICardActionHandler
+import io.adaptivecards.renderer.registration.CardRendererRegistration
+
+open class TableCellRenderArgs(renderArgs: RenderArgs, val rowIndex: Int, val colIndex: Int, val table: Table, val tableLayout: TableLayout) : RenderArgs(renderArgs)
+
+object TableCellRenderer : BaseCardElementRenderer() {
+    override fun render(renderedCard: RenderedAdaptiveCard, context: Context, fragmentManager: FragmentManager, viewGroup: ViewGroup, baseCardElement: BaseCardElement, cardActionHandler: ICardActionHandler?, hostConfig: HostConfig, renderArgs: RenderArgs): View {
+        if (renderArgs !is TableCellRenderArgs) throw IllegalArgumentException("renderArgs must be instance of TableCellRenderArgs")
+
+        val row = renderArgs.table.GetRows()[renderArgs.rowIndex]
+        val col = renderArgs.table.GetColumns()[renderArgs.colIndex]
+
+        val cell = Util.castTo(baseCardElement, TableCell::class.java)
+        val cellLayout = LinearLayout(context).apply {
+            orientation = LinearLayout.VERTICAL
+            tag = TagContent(cell)
+            layoutParams = TableRow.LayoutParams(0, TableLayout.LayoutParams.MATCH_PARENT)
+        }
+
+        cellLayout.updateLayoutParams<ViewGroup.MarginLayoutParams> {
```

Comment:
if you could move this part to a new method it could be really good for readability, this render method is almost 100 lines long